### PR TITLE
Add Bash demo

### PR DESCRIPTION
**Pull request by jessehui**
Bash change: https://github.com/occlum/bash/pull/1

### CODE REVIEW

File: demos/bash/prepare_bash_demo.sh [hunk 1]
```diff
@@ -0,0 +1,35 @@
+#! /bin/bash
+set -e
+
+current=$(pwd)
+bash_src="$current/bash_for_occlum"
+bash_git="https://github.com/jessehui/bash.git"
```
Comment (guzongmin) [inline demos/bash/prepare_bash_demo.sh:6]: Please update the repo address

File: demos/bash/prepare_bash_demo.sh [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    make -j$(nproc)
+    echo "Bash is ready."
+fi
+
+if [ ! -d "$busybox_src" ]; then
+    cd $current
+    busybox_config="$current/../fish/.config"
+    git clone -b 1_31_1 --depth 1 https://github.com/mirror/busybox.git
+    cd busybox
+    CROSS_COMPILE=/opt/occlum/toolchains/gcc/bin/occlum-
```
Comment (guzongmin) [inline demos/bash/prepare_bash_demo.sh:31]: Please help to confirm bash and bushbox is using same libc in this demo.

File: demos/bash/prepare_bash_demo.sh (hunk 1 above)
Comment (jessehui) [reply demos/bash/prepare_bash_demo.sh]: Done.

File: demos/bash/prepare_bash_demo.sh (hunk 2 above)
Comment (jessehui) [reply demos/bash/prepare_bash_demo.sh]: Change to glibc. Done. Thanks for your review.

**Review by guzongmin — COMMENTED** [review]
Please add this demo into https://github.com/occlum/occlum/blob/master/demos/README.md

**Review by guzongmin — APPROVED** [review]
LGTM

**Pull request merged**